docs/extensibility.md

```diff
@@ -0,0 +1,122 @@
+# Extensibility
+This guide provides an overview of extensibility options focusing on add-on extensions as the primary mechanism for the partners and customers. 
+## Overview
+SSP Framework is designed to be extensible. In the context of this guide, extensibility refers to the ability of customers and partners to both add new capabilities to the framework or platforms based on SSP as well as customize existing behavior, including ability to modify or override existing behavior.
+
+The following abstractions can be leveraged to add new features to the framework:
+
+- [Add-on](./core-concepts.md#add-on). Customers and partners can implement new add-ons which could be leveraged exactly the same way as the core add-ons (supplied by the framework).
+- [Resource Provider](./core-concepts.md#resource-provider). This construct allows to create resources that can be reused across multiple add-ons and/or teams. For example, IAM roles, VPC, hosted zone. 
+- [Cluster Provider](./cluster-providers/index.md). This construct allows creation of custom code that provisions EKS cluster with node groups. It can be leveraged to extend behavior such as control plane customization, custom settings for node groups.
+- [Team](./teams.md). This abstraction allows to create team templates for application and platform teams and set custom setting for network isolation, policies (network, security), software wiring (auto injection of proxies, team specific service mesh configuration) and other extensions pertinent to the teams. 
+
+## Add-on Extensions
+
+In a general case, implementation of an add-on is a class which implements the `ClusterAddOn` interface.
+
+```typescript
+export declare interface ClusterAddOn { 
+    id? : string;
+    deploy(clusterInfo: types.ClusterInfo): Promise<Construct> | void;
+}
+```
+
+**Note**: The add-on implementation can optionally supply the `id` attribute if the target add-on can be added to a blueprint more than once. 
+
+Implementation of the add-on is expected to be an exported class that implements the interface and supplies the implementation of the `deploy` method. In order for the add-on to receive the deployment contextual information about the provisioned cluster, region, resource providers and/or other add-ons, the `deploy` method takes the `ClusterInfo` parameter, which represents a structure defined in the SPI (service provider interface) contracts. The API for the cluster info structure is stable and provides access to the provisioned EKS cluster, scheduled add-ons (that have not been installed yet but are part of the blueprint) or provisioned add-ons and other contexts.
```

Comment:
Add link to `ClusterInfo` interface?

Reply:
Moved helm and non-helm add-on sections up.